docs/porting-example/index.rst

```diff
@@ -0,0 +1,345 @@
+
+Like the type, the list of ported methods in `module_defines` is initially
+empty and all the methods are still in `PointModuleMethods` which has
+been renamed to `PointModuleLegacyMethods`.
+
+Now all that is left is to replace the module initialization function with
+one that uses `HPy_MODINIT`:
+
+.. literalinclude:: steps/step_01_hpy_legacy.c
+    :lineno-match:
+    :start-at: HPy_MODINIT(step_01_hpy_legacy)
+
+And we're done!
+
+Note that the initialization function now takes an `HPyContext *` as an
+argument and that this `ctx` is passed as the first argument to calls to
+HPy API methods.
+
+`PyModule_Create` is replaced with `HPyModule_Create` and `PyType_FromSpec`
+is replaced by `HPyType_FromSpec`.
+
+`HPy_SetAttr_s` is used to add the `Point` class to the module. HPy requires no
+special `PyModule_AddObject` method.
+
+
+Step 02: Transition some methods to HPy
+---------------------------------------
+
+In the previous step we put in place the type and module definitions required
+to create an HPy extension module. In this step we will port some individual
+methods.
+
+Let us start by migrating `Point_traverse`. First we need to change
+`PyObject *obj` in the `PointObject` struct to `HPyField obj`:
+
+.. literalinclude:: steps/step_02_hpy_legacy.c
+    :lineno-match:
+    :start-at: typedef struct {
+    :end-at: } PointObject;
+
+`HPy` handles can only be short-lived -- i.e. local variables, arguments to
+functions or return values. `HPyField` is the way to store long-lived
+references to Python objects.
+
+Now we can update `Point_traverse`:
+
+.. literalinclude:: steps/step_02_hpy_legacy.c
+    :lineno-match:
+    :start-at: HPyDef_SLOT(Point_traverse, Point_traverse_impl, HPy_tp_traverse)
+    :end-before: // this is a method for creating a Point
+
+In the first line we used the `HPyDef_SLOT` macro to define a small structure
+that describes the slot being implemented. The first argument, `Point_traverse`,
+is the name to assign the structure to. The second, `Point_traverse_impl`,
+is the name of the function implementing the slot. The last, `HPy_tp_traverse`,
+specifies the kind of slot.
+
+This is a change from how slots are defined in the old C API. In the old API,
+the kind of slot is only specified much lower down in `Point_legacy_slots`. In
+HPy the implementation and kind are defined in one place using a syntax
+reminiscent of Python decorators.
+
+The implementation of traverse is now a bit simpler than in the old C API.
+We no longer need to visit `Py_TYPE(self)` and need only `HPy_VISIT`
+`self->obj`. HPy ensures that interpreter knows that the type of the instance
+is still referenced.
+
+Only struct members of type `HPyField` can be visited with `HPy_VISIT`, which
+is why we needed to convert `obj` to an `HPyField` before we implemented the
+HPy traverse.
+
+Note that only should *never* create a local variable of type `HPyField`
+because this local reference would not be visited by `tp_traverse`.
```

Comment:
Not sure if we should mention that here because it's not even straight forward to do that since you need to pass an owner object when storing to an `HPyField`. I would probably remove this paragraph.